XMonad/Hooks/DynamicIcons.hs

```diff
@@ -0,0 +1,141 @@
+{-# LANGUAGE TupleSections #-}
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  XMonad.Hooks.DynamicIcons
+-- Copyright   :  (c) Will Pierlot <willp@outlook.com.au>
+-- License     :  BSD3-style (see LICENSE)
+--
+-- Maintainer  :  Will Pierlot <willp@outlook.com.au>
+-- Stability   :  unstable
+-- Portability :  unportable
+--
+-- Dynamically change workspace text based on the contents of the workspace
+-----------------------------------------------------------------------------
+
+module XMonad.Hooks.DynamicIcons (
+    -- * Usage
+    -- $usage
+
+    -- * Creating Dynamic Icons
+    dynamicLogIconsWithPP, dynamicLogIconsConvert,
+
+    -- * Data Types
+    appIcon, IconSet,
+    IconConfig(..), Icon(..),
+
+    ) where
+import XMonad
+
+import qualified XMonad.StackSet as S
+import qualified Data.Map as M
+
+import XMonad.Hooks.DynamicLog
+import Data.Maybe (catMaybes)
+
+-- $usage
+--  Dynamically changes a 'Workspace's 'WorkspaceId' based on the 'Window's inside the Workspace.
+-- 'IconSet's describe which icons are shown depending on which windows fit a 'Query'.
+--
+-- To create an 'IconSet' make a 'Query' that returns a ['Icon'].
+--
+-- 'appIcon' can be used to simplify this process
+-- For example,
+--
+-- > icons :: IconSet
+-- > icons = composeAll
+-- >           [ className =? "discord" --> appIcon "\xfb6e"
+-- >           , className =? "Discord" --> appIcon "\xf268"
+-- >           , className =? "Firefox" --> appIcon "\63288"
+-- >           , className =? "Spotify" <||> className =? "spotify" --> appIcon "阮"
+-- >           ]
+--
+-- then you can add the hook to your config
+--
+-- > xmonad $ def
+-- >      { logHook = dynamicLogIconsWithPP icons xmobarPP <> myManageHook
+-- >      }
+--
+--  Here is an example of this
+--
+--  <<https://imgur.com/download/eauPNPz/Dynamic%20Icons%20in%20XMonad>>
+--  
+--  NOTE: It doesn't have to be an icon, a Nerd Font character is just the most basic instance of this functionality
```

Comment:
I think this is pretty clear

Reply:
I'd probably rephrase it to be something like
> NOTE: You may use any name or character here. One good choice is making use of NerdFont icons to show an icon representing the open applications.

Reply:
I have made it somethng slightly different. but similar

Reply:
I have now changed it to be something similar